src/arrays.rs

```diff
@@ -154,16 +154,16 @@ impl<const M: usize, const N: usize, const O: usize, const P: usize> HasLastAxis
     const SIZE: usize = P;
 }
 
-/// Something that has compile time known zero values.
-pub trait ZeroElements {
+/// Something that has a valid 0 value represented by all 0 bits.
+pub unsafe trait ZeroElements {
```

Comment:
I think I still prefer having a separate trait rather than re-use ZeroElements mainly because of naming. Would rather someone just look at the type and know it has to be valid as all 0 bits than read doc string.

ZeroElements is just intended to expose a const value that has everything 0'd out, which doesn't necessarily have to do with 0 bits (e.g. you mentioned `std::num::NonZeroU32` in the issue, which seems like a `impl ZeroElements` but NOT `impl ValidAsAllZeroBits`).